lib/puppetserver/ca/action/list.rb

```diff
@@ -214,9 +225,14 @@ def get_certs_or_csrs(settings, queried_state = nil)
           query = queried_state ? { :state => queried_state } : {}
           result = Puppetserver::Ca::CertificateAuthority.new(@logger, settings).get_certificate_statuses(query)
 
-          if result
+          if result.nil?
+            # nil returned from an errored lookup
+            return nil
+          elsif result
+            # values from a legit lookup
             return JSON.parse(result.body)
           else
+            # empty result from a legit lookup
```

Comment:
I think this will never be hit, and I think that's okay. In Ruby, everything besides `nil` or `false` is truthy, so I think all that was needed was to change the `nil` path to return `nil` instead of `empty`, and we don't need the third branch.

I think you should be able to go back to the conditional we had before and just change the `[]` to `nil`.